lib/take_exponential.lua

```diff
@@ -0,0 +1,70 @@
+local tokens_per_ms        = tonumber(ARGV[1])
+local bucket_size          = tonumber(ARGV[2])
+local new_content          = tonumber(ARGV[2])
+local tokens_to_take       = tonumber(ARGV[3])
+local ttl                  = tonumber(ARGV[4])
+local drip_interval        = tonumber(ARGV[5])
+local backoff_factor       = tonumber(ARGV[6])
+local mult_unit            = tonumber(ARGV[7])
+
+local current_time         = redis.call('TIME')
+local current_timestamp_ms = current_time[1] * 1000 + current_time[2] / 1000
+
+local current              = redis.pcall('HMGET', KEYS[1], 'd', 'r')
+
+if current.err ~= nil then
+    current = {}
+end
+
+-- calculate the time of next available token
+local last_token_ms = current[1] or 0
+local step = 0
+if current[2] then
+    step = tonumber(current[2])
```

Comment:
I get the intention here, but I find the name "step" a little bit confusing. 

`current[2]` goes downwards, so `step` will be decreasing with each call instead of increasing. Can we try to find an alternative?